guides/choosing_mono_or_multi_repo.md

```diff
@@ -0,0 +1,96 @@
+
+The decision should be reviewed periodically to ensure that it still meets the project requirements. If the project requirements change, it may be necessary to switch between a mono repo and a multi repo.
+
+## Considerations
+
+Teams should consider the following key drivers for choosing between a mono repo and a multi repo:
+- Size and complexity of the project
+- Technology choices
+- Collaboration
+- CI/CD
+- Release process
+- Commit history
+- Versioning
+- Security
+- Ownership
+
+### Size and complexity
+
+A mono repo is best suited for small, single language projects with a low number of dependencies. A multi repo is best suited for large, complex projects with multiple languages and dependencies.
+
+Larger projects can be difficult to manage in a mono repo, as the size and complexity of the codebase can make it difficult to navigate and understand. In this case, a multi repo may be a better choice, as it allows teams to work on smaller, more manageable codebases.
+
+Size of downloading the repository is also a consideration. A mono repo can be large and take a long time to download, particularly for new team members. A multi repo can be smaller and quicker to download, as each repository contains a smaller amount of code.
+
+Performance of scanning tools such as secret detection tools is also increased with smaller repositories.
+
+### Technology choices
+
+Mono repos are best suited for projects that use a single technology stack. Multi repos are best suited for projects that use multiple technology stacks.
+
+If a project uses multiple technology stacks, then it is more difficult to keep a single repository structure consistent.
+
+Multiple technologies in a single repository can also lead to conflicts between any plugins or dependencies that are used by the different technologies within the repository.
+
+### Collaboration
+
+Mono repos simplify collaboration as all the code is in one place.  This reduces the complexity of access control and visibility of code, pull requests, issues and release notes.
+
+However mono repos may also tightly coupled collaboration needs across independent teams.  This can lead to conflicts and bottlenecks in the development process.
+
+Multi repos are more difficult to collaborate as data is distributed across multiple repositories.  This can mean key data is missed which leads to risk.
+
+### CI/CD
+
+Mono repos simplify CI/CD as multiple projects can be built and tested together. This can reduce the complexity of creating and managing multiple pipelines and dependencies.
+
+However, mono repos can also lead to longer build times and increased complexity in managing dependencies, particularly if multiple technology choices are used.
+
+Additional tooling and complexity may be required to independently build and test projects within a mono repo depending on the scale.
```

Comment:
We have some examples of additional tooling used by projects, do we need to look at that and recommend the standard tooling to be used?

Reply:
I've added reference to Lerna as an example tool that can be used.  However, as the tone is more "our standard is multi repos", it felt a bit wrong to say "our standard is multi repos, but if you want to ignore that here's a tool you can use".

Maybe could include an acknowledgement that we know we do have some mono repos and our long term strategy is to reduce this, but to help those teams then tools like X can help reduce the constraint a mono repo in the short term?

What do you think?